### ISSUE TITLE
Spike: investigate upgrading to Primefaces 6.1.1

### ISSUE THREAD

**Issue by scolapasta** [IQSS/dataverse#3883]
We'd like to stay with the latest primefaces, so this is a spike issue in order to see what breaks in our current code if we upgrade. From this we can create additional issues in order to resolve what we discover (if anything).

**Comment by mheppler** [comment]
If anything is to break when upgrading, it *should* be outlined here in their [Migration Guide](https://github.com/primefaces/primefaces/wiki/Migration-Guide).

**Comment by pameyer** [comment]
Brief observations:
- xpaths to some UI elements change
- no obvious improvements xml/xhtml validation issues, but equally no obvious regressions
- didn't notice any breakage in the UI/API; but also didn't investigate exhaustively

**Comment by matthew-a-dunlap** [comment]
Using PrimeFaces 6.1, so far I've seen nothing break. There are a lot more aria tags (mainly aria-label tags). FWIW their repo points to 6.1, not 6.1.1 (6.1.4 is supposedly the latest). They may apply minor fixes to 6.1 and keep the same name as they go?

**Comment by landreev** [comment]
These are the 2 items that I thought needed to be specifically re-tested under the new PF version: 

1. The "growing" table used to display the list of datafiles on the dataset page. The PF component in question:

```
<p:dataTable id="filesTable"
                 scrollRows="25" scrollable="true" liveScroll="true" 
                 ... />
```

the idea is, if there are more than 25 files, it shows the top 25, then adds more dynamically, if the user tries to scroll down. 
The feature was still somewhat experimental. And we've seen it behave in unstable/unpredictable ways. As in, it would just stop expanding, even though there were more files in the dataset. There may be specific tickets opened with more info on the issue, I'll look. We were able to kind of work around it, by rearranging other components on the page - and then it just started working again. 
So, it's important to know if it's working with the new version. And maybe research if the actual behind the condition were specifically addressed in this version of PF. 

2. The file upload component. As the file is getting uploaded, PF will try to generate a thumbnail of the file. This is *different* from our thumbnail, that we are showing on the page. This one is generated by their javascript, meaning it's happening on the browser side; we found that it could not be disabled, and neither it could be configured to skip images past a certain size. So if a user is uploading a large number of large images, it's effectively freezing their browser. Many users then assume that it's our application/server that is "down". I found a *SUPER HACKY* way of just going into the PF jar, finding the piece of javascript where this is being done, and cutting it out, then re-packing the jar... We (sensibly) chose not to do that.
So, again, would be nice to know if they have since made this configurable.

**Comment by rbhatta99** [comment]
After testing out 6.1(because the repository primefaces provides for maven does not have 6.1.1), so far I have not seen anything break, or appear out of the ordinary. But, after talking to @mheppler , he pointed me to a selenium helper code which @raprasad wrote in 2014. I pulled that code and ran it. It had some old references, which I updated, and ran it again, and saw the test running without giving any bad results. 
Note: the test does not test everything, it tests some specific pages, and some random links. But, we can use this test to test out the UI elements of the application. Right now, it does the following(as far as I have run it):

- Login using 6 users and credentials

- go to advanced search

- go to a particular dataset

- go to some random dataset/dataverse

- logout


here is the link to repo if somebody is interested in checking this out, and making this more complete:

https://github.com/IQSS/selenium-helper/tree/v1.1

**Comment by pameyer** [comment]
xpath changes I'd mentioned were from selenium for:
- built-in user login
- go to specified dataverse page
- create dataset, fill in specified dataset fields
- hit save
- hit submit for review

Also not an exhaustive test; but didn't cause any failures I noticed.

**Comment by djbrooke** [comment]
Thanks @rbhatta99 for the quick meeting about this. @scolapasta is going to reach out to the Primeface folks, as there is possibly a fee for upgrading to a minor version (6.1.1 in this case).

**Comment by djbrooke** [comment]
Closing out this spike, as we have gathered the following information and determined next steps:

- We've identified some places where an upgrade will break things when we do upgrade, and we've identified some ways to test
- 6.1.1 is available, but needs to be built from source at github. If we moved to 6.1.1 now, we'd need to include it in our distribution. This is something that we won't pursue for now.
- We'll take the opportunity at Javaone to learn more about what's included in 6.1.1 before determining whether to wait for 6.2 (where Maven can work its magic) or if we need to include 6.1.1 as a .jar.
